tests/test_js_signing.py

```diff
@@ -495,3 +495,33 @@ def test_contentScript_static_values(self):
             tab.attach({ contentScript: 'doc.createElement("script")' });
         """)
         self.assert_failed(with_warnings=[warning])
+
+
+class TestObserverEvents(TestCase, RegexTestCase):
+    """Tests that code related to observer events trigger warnings."""
+
+    def test_on_newtab_changed(self):
+        """
+        Tests that adding an observer to the `newtab-url-changed` event triggers
+        a high signing severity warning.
+        """
+        patterns = (
+            'Services.obs.addObserver(NewTabObserver, "newtab-url-changed", false);',
+            (
+              'var foo = Cc["foo"].getService(Ci.nsIObserverService);'
+              'foo.addObserver(NewTabObserver, "newtab-url-changed", false);'
```

Comment:
I know it would add useless and strange indentation if it was "for real", but I think I'd rather have a multiline string here (and no parens). It would make it more readable ;)